internal/database/probe.go

```diff
@@ -0,0 +1,102 @@
+package database
+
+import (
+	"encoding/binary"
+	"encoding/json"
+	"fmt"
+
+	"github.com/boltdb/bolt"
+	"github.com/madjlzz/madprobe/internal/model"
+)
+
+type probeDto struct {
+	ID    int
+	Name  string
+	URL   string
+	Delay uint
+}
+
+func newProbeModel(probe model.Probe) *probeDto {
+	// ID is generated by bold db
+	// Status and Finish chan not persisted
+	return &probeDto{
+		Name:  probe.Name,
+		URL:   probe.URL,
+		Delay: probe.Delay,
+	}
+}
+
+func (p *probeDto) toProbe() model.Probe {
+	return model.Probe{
+		Name:  p.Name,
+		URL:   p.URL,
+		Delay: p.Delay,
+		ID:    p.ID,
+	}
+}
+
+// InsertProbe returns saved probe (with updated ID) and error. Executes onSuccess with inserted probe as part of the transaction.
+func (c *client) InsertProbe(probe model.Probe, onSuccess func(probe model.Probe) error) (model.Probe, error) {
+	var probeModel = newProbeModel(probe)
+	err := c.db.Update(func(tx *bolt.Tx) error {
+		b := tx.Bucket([]byte(probeBucket))
+		id, _ := b.NextSequence()
+		probeModel.ID = int(id)
```

Comment:
You can use directly the `name` as a unique identifier for the DB.

Reply:
sure